### PR TITLE
[Early sketch]

### PR DESCRIPTION
**Pull request by kbara**

### CODE REVIEW

File: src/apps/config/channel.lua [hunk 1]
```diff
@@ -27,7 +27,7 @@ local ring_buffer_t = ffi.typeof([[struct {
 -- A: We need a variable-sized mapping.
 local function create_ring_buffer (name, size)
    local path = shm.resolve(name)
-   shm.mkdir(path)
+   shm.mkdir(lib.dirname(path))
```
Comment (wingo) [inline src/apps/config/channel.lua:30]: This fix should apply to the merge in a separate patch, just to get the test suite working after the merge with no changes of ours on top

File: src/program/lwaftr/setup.lua [hunk 2]
```diff
@@ -451,9 +453,14 @@ function reconfigurable(f, graph, conf, ...)
       f(graph, conf, unpack(args))
       return graph
    end
-   config.app(graph, 'leader', leader.Leader,
-              { setup_fn = setup_fn, initial_configuration = conf,
-                follower_pids = { require('syscall').getpid() },
-                schema_name = 'snabb-softwire-v1'})
-   config.app(graph, "follower", follower.Follower, {})
+   local follower_pid = S.fork()
```
Comment (wingo) [inline src/program/lwaftr/setup.lua]: I think here the expectation is that local follower_id = worker.start(...).

File: src/core/worker.lua [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+      -- Restart the process with execve().
+      -- /proc/$$/exe is a link to the same Snabb executable that we are running
+      local env = {}
+      for key, value in pairs(S.environ()) do
+         table.insert(env, key.."="..value)
+      end
+      S.execve(("/proc/%d/exe"):format(S.getpid()), {}, env)
+   else
+      -- Parent process
+      children[name] = { pid = pid }
```
Comment (wingo) [inline src/core/worker.lua:52]: Is this the supervisor of the worker or the worker itself?  We need a way to get the worker PID.

Comment (kbara) [reply src/core/worker.lua:52]: It's the supervisor of the worker.

**Pull request closed without merging**